scripts/morty/isobar_morty_pde.py

```diff
@@ -0,0 +1,502 @@
+        self.nu_vec = self._format_spatial(nu1, nu2)
+
+        self.mu = {}
+        self.mu['a'] = self._format_spatial((lams['a'] + losses['1a']),
+                                            (lams['a'] + losses['2a']))
+        self.mu['b'] = self._format_spatial((lams['b'] + losses['1b']),
+                                            (lams['b'] + losses['2b']))
+        self.mu['c'] = self._format_spatial((lams['c'] + losses['1c']),
+                                            (lams['c'] + losses['2c']))
+        self.mu['d_m1'] = self._format_spatial((lams['d_m1']
+                                                + losses['1d_m1']),
+                                               (lams['d_m1'] + losses['2d']))
+        self.mu['d'] = self._format_spatial((lams['d'] + losses['1d']),
+                                            (lams['d'] + losses['2d']))
+        self.S = {}
+        self.S['a'] = self._format_spatial((FYs['a']/vol1), (0/vol2))
+
+        self.lama = lams['a']
+        self.lamb = lams['b']
+        self.lamc = lams['c']
+        self.lamd = lams['d']
+        self.lamd_m1 = lams['d_m1']
+
+        self.FYb = FYs['b']
+        self.FYc = FYs['c']
+        self.FYd = FYs['d']
+        self.FYd_m1 = FYs['d_m1']
+
+        self.br_c_d = br_c_d
+        self.br_dm1_d = br_dm1_d
+
+        self.vol1 = vol1
+        self.vol2 = vol2
+
+        return
+
+    def _external_PDE_no_step(self, conc, isotope):
+        """
+        This function applies a single time step iteration of the PDE
+
+        Parameters
+        ----------
+        conc : 1D vector
+            Concentration over spatial nodes at previous time
+        isotope : string
+            Nuclide isobar indicator (a, b, c, d, or d_m1)
+
+        Returns
+        -------
+        conc : 1D vector
+            Concentration over spatial nodes at current time
+        """
+        S_vec = self.S[isotope]
+        mu_vec = self.mu[isotope]
+        J = np.arange(0, self.nodes)
+        Jm1 = np.roll(J,  1)
+        dz = np.diff(self.zs)[0]
+
+        conc_mult = 1 - mu_vec * self.dt
+        add_source = S_vec * self.dt
+        lmbda = (self.nu_vec * self.dt / dz)
+        conc = add_source + conc_mult * conc + lmbda * (conc[Jm1] - conc)
+
+        return conc
+
+    def _initialize_result_mat(self, PDE=True):
+        """
+        Set up the 3D result matrix with the form
+            time, space, nuclide
+            with 5 nucldies in the isobar available
+
+        Parameters
+        ----------
+        PDE : bool
+            If using PDE True, False for ODE
+
+        Returns
+        -------
+        result_mat : 3D matrix (2D if PDE False)
+            Holds values over time, space, and nuclide (in that order)
+
+        """
+        if PDE:
+            nodes = self.nodes
+        else:
+            nodes = 1
+        result_mat = np.zeros((len(self.ts), nodes, 5))
+        self.conc_a = np.array([0] * nodes)
```

Comment:
```suggestion
        result_mat = np.zeros((len(self.ts), nodes, 5))
        self.conc_a = np.zeros(nodes)
```

This causes me great confusion. You know that `np.zeros` exists, obviously.

I can't make a suggestion for every item, but please fix this.